pysph/tools/interpolator.py

```diff
@@ -27,12 +30,134 @@ def post_loop(self, d_idx, d_prop, d_number_density):
             d_prop[d_idx] /= d_number_density[d_idx]
 
 
-class InterpolateSPH(Equation):
-    def initialize(self, d_idx, d_prop):
-        d_prop[d_idx] = 0.0
+class UpdateD(Equation):
```

Comment:
A nicer name is necessary, I cannot figure out what the intent of this name is.

Reply:
A docstring on how this formula was arrived at or a link to a suitable reference is important.

Reply:
added refs and changed var names